src/Playwright/Transport/Converters/EvaluateArgumentValueConverter.cs

```diff
@@ -283,7 +291,9 @@ private static object ParseEvaluateResult(JsonElement result, Type t, JsonSerial
 
             if (t == typeof(JsonElement?))
             {
-                return result;
+                // we need to make sure that the returned instance *is not* dependent on the
+                // lifecycle of JsonDocument, hence we Clone it
+                return result.Clone();
```

Comment:
How about making clone of the element as early as possible, i.e. at line 50 (`var result = document.RootElement; => var result = document.RootElement.Clone();`) so that we don't have to think about it everywhere?